### PR TITLE
TDD project

### PR DESCRIPTION
**Pull request by Abenezer-Tilahun**
# Description

- In this project, we implemented TDD project.

# General requirements

- There are no linter errors.
- Followed the GitFlow.
- The documentation is professional.

# Ruby requirements

- Followed best practices for Ruby

# Project requirements

- [x] Create a class called Solver.
- [x] Create a method called factorial that takes one argument, an integer N, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to N and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.
- [x] Create a method called reverse that takes one argument, a string word, and returns word reversed (e.g. if word is "hello" it returns "olleh").
- [x] Create a method called fizzbuzz that takes one argument, an integer N, and returns a string. The returned string is constructed following these rules:
- When N is divisible by 3, return "fizz".
- When N is divisible by 5, return "buzz".
- When N is divisible by 3 and 5, return "fizzbuzz".
- Any other case, return N as a string (e.g. say N is 7 then return "7").

- [x] Start with tests for all of the methods described above. Your commit history should prove that you have used TDD

### CODE REVIEW

File: solver.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    word.reverse
+  end
+
+  def fizzbuzz(number)
+    solver = ''
+
+    solver += 'fizz' if (number % 3).zero?
+    solver += 'buzz' if (number % 5).zero?
+    return number if solver.strip.empty?
+
```
Comment (Gambit142) [inline solver.rb]: - [x] It should return the string version of the number if it is not divisible by 3 or 5. Currently, it returns the number as it is. Kindly fix this. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Abenezer-Tilahun @vikipretium 👋 , 

Good job team!

## Highlights
- Followed TDD pattern 💯 
- Has `fizzbuzz` method 💯 
- Has `factorial` method 💯 
- Has `reverse` method 💯 

However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: solver.rb (hunk 1 above)
Comment (Abenezer-Tilahun) [reply solver.rb]: @Gambit142 Thanks for reviewing it, We fix the issues.

Comment (vikipretium) [reply solver.rb]: fixed it @Gambit142 👍🏼

**Review by julie-ify — APPROVED** [review]
Hi @Abenezer-Tilahun and @vikipretium, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**